### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by TimVoigtlaender**
Rework parsing of needed envs in setup script to be more stable. Add more tests in setup script. Add domain to etp containers. Fix bug in upload of env tarballs if env is not available in cvmfs.

### CODE REVIEW

File: scripts/ParseNeededEnv.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    base_env = config["DEFAULT"]["ENV_NAME"]
+except (KeyError, configparser.ParsingError) as error:
+    print("File at {} is not a valid config file.@{}".format(cfg_path, error))
+    exit(1)
+
+all_env = [base_env]
+for section in config.sections():
+    all_env.append(config[section]["ENV_NAME"])
+all_env = list(set(all_env))
+print(base_env)
```
Comment (harrypuuter) [inline scripts/ParseNeededEnv.py:24]: this printout is too much, breaks the setup script ?

File: scripts/ParseNeededEnv.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+if not os.path.isfile(cfg_path):
+    print("File was not found at {}".format(cfg_path))
+    exit(1)
+config = configparser.ConfigParser()
+
+try:
+    config.read(cfg_path)
+    base_env = config["DEFAULT"]["ENV_NAME"]
+except (KeyError, configparser.ParsingError) as error:
+    print("File at {} is not a valid config file.@{}".format(cfg_path, error))
```
Comment (harrypuuter) [inline scripts/ParseNeededEnv.py:17]: Change to a more meaningful error message to state what is missing

**Review by harrypuuter — CHANGES_REQUESTED** [review]
Two things I found that should be changed

File: scripts/ParseNeededEnv.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    exit(1)
+
+all_env = [base_env]
+for section in config.sections():
+    all_env.append(config[section]["ENV_NAME"])
```
Comment (harrypuuter) [inline scripts/ParseNeededEnv.py:22]: .strip() here to be save ?

File: setup.sh [hunk 4]
```diff
@@ -84,12 +97,13 @@ action() {
             # Install conda if necessary
             if [ ! -f "miniconda/bin/activate" ]; then
                 # Miniconda version used for all environments
-                MINICONDA_VERSION="Miniconda3-py39_4.10.3-Linux-x86_64"
+                # MINICONDA_VERSION="Miniconda3-py39_4.10.3-Linux-x86_64"
+                MINICONDA_VERSION="Miniconda3-latest-Linux-x86_64"
```
Comment (harrypuuter) [inline setup.sh:101]: I would be in favor of pinning this version, instead of using latest, to avoid breaking when setting up a new env

Comment (harrypuuter) [reply setup.sh:101]: I am setting up the updated env with `23.5.2` so we should pin that version using `Miniconda3-py39_23.5.2-0-Linux-x86_64`

File: scripts/ParseNeededEnv.py (hunk 1 above)
Comment (TimVoigtlaender) [reply scripts/ParseNeededEnv.py:24]: No, this is intended to deal with setups that use more than one env across their tasks. The first print provides the env that is supposed to be used for the local setup. Everything after the first is the full list of envs that will be needed and therefore set up. 
I could make it a bit shorter by only providing the second half and putting the starting env at the start of that list.

Comment (harrypuuter) [reply scripts/ParseNeededEnv.py:24]: for me, it did not work with KingMaker and both printouts

Comment (TimVoigtlaender) [reply scripts/ParseNeededEnv.py]: Reworked in commit. Please let me know if issues persist on your side.

**Review by harrypuuter — APPROVED** [review]

**Pull request merged**